diff --git a/dist/client.js b/dist/client.js
index d47337812c5f5dac1e2261f13050caa20b06bf47..966f95696bf2364e2d7fc772418775ce758a5d17 100644
--- a/dist/client.js
+++ b/dist/client.js
@@ -3360,8 +3360,9 @@ handlers.set("GUILD_DELETE", async (client, d) => {
   const g = d;
   const guild = client.guilds.get(g.id);
   if (guild) {
-    client.guilds.delete(g.id);
-    client.emit(Events.GuildDelete, guild);
+    const unavailable = g.unavailable === true;
+    if (!unavailable) client.guilds.delete(g.id);
+    client.emit(Events.GuildDelete, guild, unavailable);
   }
 });
 handlers.set("CHANNEL_CREATE", async (client, d) => {
diff --git a/dist/client.mjs b/dist/client.mjs
index 16eca315fc0a7be5c122e422efc7f127ee58626d..c65adc1c6c3fb649f5b2e862cf10b1953e3ce6d2 100644
--- a/dist/client.mjs
+++ b/dist/client.mjs
@@ -3350,8 +3350,9 @@ handlers.set("GUILD_DELETE", async (client, d) => {
   const g = d;
   const guild = client.guilds.get(g.id);
   if (guild) {
-    client.guilds.delete(g.id);
-    client.emit(Events.GuildDelete, guild);
+    const unavailable = g.unavailable === true;
+    if (!unavailable) client.guilds.delete(g.id);
+    client.emit(Events.GuildDelete, guild, unavailable);
   }
 });
 handlers.set("CHANNEL_CREATE", async (client, d) => {
diff --git a/dist/index.js b/dist/index.js
index 234bb568aa8c72d410ef8d07797b0d1c92024164..ea6f1f12aecadce4f3fc05b444fcc1b739b0e5cf 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -3437,8 +3437,9 @@ handlers.set("GUILD_DELETE", async (client, d) => {
   const g = d;
   const guild = client.guilds.get(g.id);
   if (guild) {
-    client.guilds.delete(g.id);
-    client.emit(Events.GuildDelete, guild);
+    const unavailable = g.unavailable === true;
+    if (!unavailable) client.guilds.delete(g.id);
+    client.emit(Events.GuildDelete, guild, unavailable);
   }
 });
 handlers.set("CHANNEL_CREATE", async (client, d) => {
diff --git a/dist/index.mjs b/dist/index.mjs
index 1aa1b631a6179bf24aae217e7e52d2207e86a4b2..efd1ed24d28fa2818b0620a047ce42ca2fe9e21b 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -3370,8 +3370,9 @@ handlers.set("GUILD_DELETE", async (client, d) => {
   const g = d;
   const guild = client.guilds.get(g.id);
   if (guild) {
-    client.guilds.delete(g.id);
-    client.emit(Events.GuildDelete, guild);
+    const unavailable = g.unavailable === true;
+    if (!unavailable) client.guilds.delete(g.id);
+    client.emit(Events.GuildDelete, guild, unavailable);
   }
 });
 handlers.set("CHANNEL_CREATE", async (client, d) => {
diff --git a/dist/subpath-message-DSs_mdc4.d.mts b/dist/subpath-message-DSs_mdc4.d.mts
index 08e110968fa64f1f81d51d994b67dc4746bf6ba9..772e1ab22efbfea45123508cff72c2ef854bfe4e 100644
--- a/dist/subpath-message-DSs_mdc4.d.mts
+++ b/dist/subpath-message-DSs_mdc4.d.mts
@@ -878,8 +878,9 @@ declare class Guild extends Base {
      * Requires Manage Channels permission.
      */
     createChannel(data: {
-        type: 0 | 2 | 4 | 5;
+        type: 0 | 2 | 4 | 998;
         name: string;
+        url?: string | null;
         parent_id?: string | null;
         topic?: string | null;
         bitrate?: number | null;
@@ -1964,7 +1965,7 @@ interface ClientEvents {
     [Events.InteractionCreate]: [interaction: AnyInteraction];
     [Events.GuildCreate]: [guild: Guild];
     [Events.GuildUpdate]: [oldGuild: Guild, newGuild: Guild];
-    [Events.GuildDelete]: [guild: Guild];
+    [Events.GuildDelete]: [guild: Guild, unavailable: boolean];
     [Events.ChannelCreate]: [channel: GuildChannel];
     [Events.ChannelUpdate]: [oldChannel: Channel, newChannel: Channel];
     [Events.ChannelDelete]: [channel: Channel];
diff --git a/dist/subpath-message-DSs_mdc4.d.ts b/dist/subpath-message-DSs_mdc4.d.ts
index 08e110968fa64f1f81d51d994b67dc4746bf6ba9..772e1ab22efbfea45123508cff72c2ef854bfe4e 100644
--- a/dist/subpath-message-DSs_mdc4.d.ts
+++ b/dist/subpath-message-DSs_mdc4.d.ts
@@ -878,8 +878,9 @@ declare class Guild extends Base {
      * Requires Manage Channels permission.
      */
     createChannel(data: {
-        type: 0 | 2 | 4 | 5;
+        type: 0 | 2 | 4 | 998;
         name: string;
+        url?: string | null;
         parent_id?: string | null;
         topic?: string | null;
         bitrate?: number | null;
@@ -1964,7 +1965,7 @@ interface ClientEvents {
     [Events.InteractionCreate]: [interaction: AnyInteraction];
     [Events.GuildCreate]: [guild: Guild];
     [Events.GuildUpdate]: [oldGuild: Guild, newGuild: Guild];
-    [Events.GuildDelete]: [guild: Guild];
+    [Events.GuildDelete]: [guild: Guild, unavailable: boolean];
     [Events.ChannelCreate]: [channel: GuildChannel];
     [Events.ChannelUpdate]: [oldChannel: Channel, newChannel: Channel];
     [Events.ChannelDelete]: [channel: Channel];
